tests/smoke_test.go

```diff
@@ -785,6 +785,50 @@ func TestRefreshUpgradeWarning(t *testing.T) {
 
 	// Run a refresh and check that the warning is shown
 	stdout, _ := e.RunCommand("pulumi", "refresh", "--yes")
-	assert.Contains(t, stdout, "refresh operation is using an older version of plugin 'random' "+
+	assert.Contains(t, stdout, "refresh operation is using an older version of package 'random' "+
 		"than the specified program version: 4.13.0 < 4.16.7")
 }
+
+// Similar to TestRefreshUpgradeWarning above, but test that this correctly picks up changes to paramaterised packages.
+//
+//nolint:paralleltest // pulumi new is not parallel safe
```

Comment:
Wonder if it's worth making this one a destroy just to eek out the extra confidence across both the commands?